### PR TITLE
feat(behavior_path_planner): weakened noise filtering of drivable area

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

The iteration of closing drivable area will be 1, which was previously 10.
If the iteration is 10, there is a certain difference between drivable area's boundary and lane's boundary.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utilities.cpp [hunk 1]
```diff
@@ -1227,7 +1227,7 @@ OccupancyGrid generateDrivableArea(
     }
 
     // Closing
-    constexpr int num_iter = 10;  // TODO(Horibe) Think later.
+    constexpr int num_iter = 1;  // TODO(Horibe) Think later.
     cv::Mat cv_erode, cv_dilate;
     cv::erode(cv_image, cv_erode, cv::Mat(), cv::Point(-1, -1), num_iter);
```
Comment (taikitanaka3) [inline planning/behavior_path_planner/src/utilities.cpp:1232]: @takayuki5168 
[FYI] you can replace erode and dilate processing to morphologyEx instead if you want to
[cv::morphologyEx](https://docs.opencv.org/3.4/d4/d86/group__imgproc__filter.html#ga67493776e3ad1a3df63883829375201f)

Comment (takayuki5168) [reply planning/behavior_path_planner/src/utilities.cpp:1232]: Thank you. In this PR, to be simple, I will keep the current change.

**Comment by TakaHoribe** [comment]
@takayuki5168 Originally, the drivable area extension was implemented as "expand into a vertical direction of the lenelet's centerline" which causes the unexpected drivable area. 
image

The implementation has already been improved, and no concern exists in my side.

**Comment by takayuki5168** [comment]
Side shift uses "expand into a vertical direction of the lenelet's centerline" as Horibe-san commented.
This implementation does not guarantee the connection without any longitudinal space of two successive lanelets as follows.
image
image

**Comment by takayuki5168** [comment]
So I added this commit to modify front and back points of the expanded lanelet so that the successive lanelets will not have any longitudinal space between them.
https://github.com/autowarefoundation/autoware.universe/pull/838/commits/cc5bfc829f4d04702776fd03b302ba609ecba810

The result were as follows, which seems fine.
image
image

**Review by TakaHoribe — APPROVED** [review]
The code LGTM. Thank you!

**Comment by TakaHoribe** [comment]
@takayuki5168 Sorry, just one thing before merge, the "TODO" comment needs to be removed/modified. Also please add a brief for why we have the erode and dilate.

**Comment by takayuki5168** [comment]
@TakaHoribe Ok, added
https://github.com/autowarefoundation/autoware.universe/pull/838/commits/8f6b8620129f20be5a1eb7e731e1ce8c5195b892

**Pull request merged**